pkg/prometheus/agent/statefulset.go

```diff
@@ -155,48 +154,24 @@ func makeStatefulSetSpec(
 ) (*appsv1.StatefulSetSpec, error) {
 	cpf := p.GetCommonPrometheusFields()
 
-	pImagePath, err := operator.BuildImagePath(
-		operator.StringPtrValOrDefault(cpf.Image, ""),
-		operator.StringValOrDefault("", c.PrometheusDefaultBaseImage),
-		operator.StringValOrDefault(cpf.Version, operator.DefaultPrometheusVersion),
-		"",
-		"",
-	)
+	pImagePath, err := buildImagePath(cpf, c)
 	if err != nil {
 		return nil, err
 	}
 
 	cpf.EnableFeatures = append(cpf.EnableFeatures, "agent")
```

Comment:
reading this, I realize that someone could define `.spec.enabledFeatures: [agent]`  in a PrometheusAgent object. It doesn't do any harm (Prometheus will start and log E`xperimental agent mode enabled.` twice) but we might want to "fix" it.

Reply:
So we need to check if `cpf.EnableFeatures` already contains `agent` or not right? Should I fix this in a separate PR?